architecture-design.md

```diff
@@ -34,7 +41,17 @@ apiRoutes.js sets up one route for the app to make an ajax request to. The route
-It begins by writing a static html string that contains the head of the html page and the start of the body. It then uses a method provided by react-dom/server called renderToNodeStream. renderToNodeStream takes the react application, imported from client/components/app, wraps it in a StaticRouter. The app is wrapped in a StaticRouter because this is a single-page application; regardless of what route a user attempts to go to while making a request to the talent domain (other than /api/learners), the express server will always have the same response, so the app is sent in a StaticRouter, as the user will never be going to a different route. The app is also wrapped in a Provider; the Provider is a component provided by react-router, and it allows your react application to make use of all the features that Redux provides. We will be going into more detail about what this entails later on. After writing the initial head and body of the html document, the express server will then send over the React app as a stream in chunks. Line 44 is necessary because it tells the browser that express is not yet done streaming after all of the chunks have been sent. This is necessary because on line 46, we send the final portion of the html document, which sends the bundle file created by webpack as a script to the browser, and closes off the html body and document. It then notifies the browser on line 47 that the server has finished its response.
+It begins by writing a static html string that contains the head of the html page and the start of the body. It then uses a method provided by react-dom/server called renderToNodeStream. renderToNodeStream takes the react application, imported from client/components/app, wraps it in a StaticRouter. The app is wrapped in a StaticRouter because this is a single-page application; regardless of what route a user attempts to go to while making a request to the talent domain (other than /api/learners), the express server will always have the same response, so the app is sent in a StaticRouter, as the user will never be going to a different route. The app is also wrapped in a Provider; the Provider is a component provided by react-router, and it allows your react application to make use of all the features that Redux provides. We will be going into more detail about what this entails later on. After writing the initial head and body of the html document, the express server will then send over the React app as a stream in chunks. Line 46 is necessary because it tells the browser that express is not yet done streaming after all of the chunks have been sent. This is necessary because on line 48, we send the final portion of the html document, which sends the bundle file created by webpack as a script to the browser, and closes off the html body and document. It then notifies the browser on line 47 that the server has finished its response.
```

Comment:
in the future, consider breaking up text into bullet points, so smaller edits can be seen line-by-line